### PR TITLE
[ios, macos] Fix conditional expression parsing error.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Fixes a parsing issue when conditional nested expressions are passed as parameters of `expressionForConditional`.

cc @riastrad

### CODE REVIEW

File: platform/darwin/src/NSExpression+MGLAdditions.mm [hunk 1]
```diff
@@ -1230,27 +1230,10 @@ - (id)mgl_jsonExpressionObject {
         }
             
         case NSConditionalExpressionType: {
-            NSMutableArray *arguments = [NSMutableArray arrayWithObjects:self.predicate.mgl_jsonExpressionObject, nil];
+            NSMutableArray *arguments = [NSMutableArray arrayWithObjects:@"case", self.predicate.mgl_jsonExpressionObject, nil];
+            [arguments addObject:self.trueExpression.mgl_jsonExpressionObject];
+            [arguments addObject:self.falseExpression.mgl_jsonExpressionObject];
             
-            if (self.trueExpression.expressionType == NSConditionalExpressionType) {
-                // Fold nested conditionals into a single case expression.
```
Comment (1ec5) [inline platform/darwin/src/NSExpression+MGLAdditions.mm]: It is valid to fold nested conditionals into a single [`case` expression](https://docs.mapbox.com/mapbox-gl-js/style-spec/#expressions-case); that leads to a more succinct expression. However, the problem is that the nested conditional’s false argument somehow made it into the overall expression. Nesting should only be possible in this situation:

> if foo then bar else (if baz then boo else quux)

but we were also flattening the following situation:

> if foo then (if bar then baz else boo) else quux

In other words, it’s specifically this if statement causing the bug, not the subsequent if statement for the false case.

In any case, the change is correct, just less streamlined. I don’t know if there are performance implications for very deeply nested case expressions, but it does result in something unwieldy if the developer wants to use this method directly for some reason.
